internal/provider/resource_gpg_key_test.go

```diff
@@ -0,0 +1,168 @@
+GbFOWUM9+UAjQfV6tnQfkvEFAmPipfcCGwwFCQPCZwAACgkQQfV6tnQfkvGnbAwA
+uMZ4ThOXOA17iyBgKQ4tj0TGTqErKb0dxuuvf0g+ozRfFdnhr+UiuD2QtgNcYNNm
+U+qLAt96sPCN+nit2/coE0P+YI24iTC8AYJXXSgP/ZnyjkkbKNQEBRm/hdocejzB
+5BM3ztV1VriQqIQEqp4HTzcOTXiEhZ8jZW0mrBTlHenYMe/83zoBuABQGMnuy/JJ
+pSgJ+XQ6uBnGa7b/35nHUfoIhC2GNQ8/uI/VBy1vhnEBFubROVMyss9IpTheDHOC
+oYbE8Lq9J8Giu8mqyF4ifzXl9A2lowPFDg6Ey9Yms+wnVWUD2uMdQ00PIMB0HUFo
+alugyNSEqc6GP9rOUkR4TUwNmeV1OJCJtX6sdb+WY2ZczoiT7SYVBkqS6xEujeRy
+DGGMeh5+2/26EiP2nBcIJqTCqZi+yq/5k7QKNtNYNdb/u1WvtseDsfOgekZSwOoN
+lNBLBcAMCdEMd4qgt0YvzKzE3GbQoiAkBKJ2qoqun2MXM60324j01B/x/r3E+p15
+=HJT6
+-----END PGP PUBLIC KEY BLOCK-----
+`
+	createFingerprint := "C48F7F19B14E59433DF9402341F57AB6741F92F1"
+	createGPGKeyID := "41F57AB6741F92F1"
+	createGroupPath := testGroupPath
+
+	resource.Test(t, resource.TestCase{
+
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+
+			// Create and read back a GPG key.
+			{
+				Config: testGPGKeyConfigurationCreate(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Verify values that should be known.
+					resource.TestCheckResourceAttr("tharsis_gpg_key.tgk", "ascii_armor", createASCIIArmor),
+					resource.TestCheckResourceAttr("tharsis_gpg_key.tgk", "fingerprint", createFingerprint),
+					resource.TestCheckResourceAttr("tharsis_gpg_key.tgk", "gpg_key_id", createGPGKeyID),
+					resource.TestCheckResourceAttr("tharsis_gpg_key.tgk", "group_path", createGroupPath),
+
+					// Verify dynamic values have any value set in the state.
+					resource.TestCheckResourceAttrSet("tharsis_gpg_key.tgk", "id"),
+					resource.TestCheckResourceAttrSet("tharsis_gpg_key.tgk", "last_updated"),
+					resource.TestCheckResourceAttrSet("tharsis_gpg_key.tgk", "created_by"),
+				),
+			},
+
+			// Import the state.
+			{
+				ResourceName:      "tharsis_gpg_key.tgk",
+				ImportStateId:     "FIXME: What/which ID should this be?", // FIXME: What/which ID should this be?
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+
+			// Update is allowed _ONLY_ if nothing changes, so use the same configuration as create.
+			{
+				Config: testGPGKeyConfigurationCreate(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Verify values that should be known.
+					resource.TestCheckResourceAttr("tharsis_gpg_key.tgk", "ascii_armor", createASCIIArmor),
+					resource.TestCheckResourceAttr("tharsis_gpg_key.tgk", "fingerprint", createFingerprint),
+					resource.TestCheckResourceAttr("tharsis_gpg_key.tgk", "gpg_key_id", createGPGKeyID),
+					resource.TestCheckResourceAttr("tharsis_gpg_key.tgk", "group_path", createGroupPath),
+
+					// Verify dynamic values have any value set in the state.
+					resource.TestCheckResourceAttrSet("tharsis_gpg_key.tgk", "id"),
+					resource.TestCheckResourceAttrSet("tharsis_gpg_key.tgk", "last_updated"),
+					resource.TestCheckResourceAttrSet("tharsis_gpg_key.tgk", "created_by"),
+				),
+			},
+
```

Comment:
You could add a test case here similar to Managed Identity Aliases to verify the key can't be updated:
https://github.com/martian-cloud/terraform-provider-tharsis/blob/6146ae25f50e6509ab80d919a69061ba0e453537/internal/provider/resource_managed_identity_alias_test.go#L65